src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,556 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+from prog_models.prognostics_model import PrognosticsModel
+import prog_models.models.uav_model.trajectory.route as route 
+import prog_models.models.uav_model.trajectory.trajectory as trajectory
+from prog_models.models.uav_model.vehicles import AircraftModels
+from prog_models.models.uav_model.utilities import geometry
+
+import numpy as np
+import datetime
+from copy import deepcopy
+import prog_models.models.uav_model.utilities.geometry as geom
+from warnings import warn
+from prog_models.exceptions import ProgModelInputException
+
+class UAVGen(PrognosticsModel):
+    """
+    Vectorized prognostics :term:`model` to generate a predicted trajectory for a UAV. 
+    
+    Model generates cartesian positions and velocities, pitch, roll, and yaw, and angular velocities throughout time to satisfy some user-define waypoints. 
+
+    :term:`Events<event>`: (1)
+        TrajectoryComplete: percentage of waypoints remaining for trajectory to pass through
```

Comment:
The event would actually be completion of all waypoints, then the event state is the percentage of waypoints completed.

```suggestion
        TrajectoryComplete: All waypoints are completed
```